synthesizer/src/program/instruction/operation/literals.rs

```diff
@@ -84,19 +105,32 @@ impl<N: Network, O: Operation<N, Literal<N>, LiteralType, NUM_OPERANDS>, const N
             inputs.iter().map(|input| RegisterType::Plaintext(PlaintextType::from(input.to_type()))).collect();
 
         // Compute the operation.
-        let output = O::evaluate(&inputs.try_into().map_err(|_| anyhow!("Failed to prepare operands in evaluate"))?)?;
+        let outputs = O::evaluate(&inputs.try_into().map_err(|_| anyhow!("Failed to prepare operands in evaluate"))?)?;
         // Compute the output type.
-        let output_type = RegisterType::Plaintext(PlaintextType::from(output.to_type()));
+        let output_types = outputs
+            .iter()
+            .map(|output| RegisterType::Plaintext(PlaintextType::from(output.to_type())))
+            .collect::<Vec<_>>();
```

Comment:
```suggestion
            .map(|output| RegisterType::Plaintext(PlaintextType::from(output.to_type())));
```
seems like we only need to iterate over them later, so this allocation is avoidable